src/openrct2/actions/PlacePeepSpawnAction.hpp

```diff
@@ -98,24 +98,34 @@ DEFINE_GAME_ACTION(PlacePeepSpawnAction, GAME_COMMAND_PLACE_PEEP_SPAWN, GameActi
         res->Expenditure = ExpenditureType::LandPurchase;
         res->Position = _location;
 
-        // If we have reached our max peep spawns, remove the oldest spawns
-        while (gPeepSpawns.size() >= MAX_PEEP_SPAWNS)
-        {
-            PeepSpawn oldestSpawn = *gPeepSpawns.begin();
-            gPeepSpawns.erase(gPeepSpawns.begin());
-            map_invalidate_tile_full(oldestSpawn);
-        }
-
         // Shift the spawn point to the edge of the tile
         auto spawnPos = CoordsXY{ _location.ToTileCentre() }
             + CoordsXY{ DirectionOffsets[_location.direction].x * 15, DirectionOffsets[_location.direction].y * 15 };
 
-        // Set peep spawn
         PeepSpawn spawn;
         spawn.x = spawnPos.x;
         spawn.y = spawnPos.y;
         spawn.z = _location.z;
         spawn.direction = _location.direction;
+
+        // Don't allow multiple peep spawns on same tile
+        if (!gPeepSpawns.empty())
+        {
+            if (std::find(std::begin(gPeepSpawns), std::end(gPeepSpawns), spawn) != std::end(gPeepSpawns))
+            {
+                return std::make_unique<GameActionResult>(GA_ERROR::UNKNOWN, STR_ERR_CANT_PLACE_PEEP_SPAWN_HERE);
```

Comment:
Rather than showing an error, it would be very handy if we could remove peep spawns by clicking existing ones. Currently there is no way to remove them easily.

Reply:
That's above my capabilities (yet) 😃

Reply:
It shouldn't be too difficult. I can take a look at it, but I can't promise any timescale for it.

Reply:
I have now made the change in question.